XMonad/Layout/BinarySpacePartition.hs

```diff
@@ -425,13 +425,14 @@ resizeSplit :: Direction2D -> (Rational,Rational) -> Zipper Split -> Maybe (Zipp
 resizeSplit _ _ z@(_, []) = Just z
 resizeSplit dir (xsc,ysc) z = case goToBorder dir z of
   Nothing -> Just z
-  Just (t, crumb) -> Just $ case dir of
+  Just (t@Node{}, crumb) -> Just $ case dir of
     R -> (t{value=sp{ratio=scaleRatio (ratio sp) xsc}}, crumb)
     D -> (t{value=sp{ratio=scaleRatio (ratio sp) ysc}}, crumb)
     L -> (t{value=sp{ratio=1-scaleRatio (1-ratio sp) xsc}}, crumb)
     U -> (t{value=sp{ratio=1-scaleRatio (1-ratio sp) ysc}}, crumb)
     where sp = value t
           scaleRatio r fac = min 0.9 $ max 0.1 $ r*fac
+  Just (Leaf{}, _) -> undefined
```

Comment:
Is there any harm in just returning `Nothing` here?  If the `undefined` stays, I think it would be worth adding the commit message as a comment there, so people will know this is intentional

Reply:
The point of the undefined is to silence the warning and to produce an error message annotated with source code location if the surrounding code changes in a way that this can match. Returning Nothing may have consequences depending on how the surrounding code changes, and it's hard to foresee what consequences (although maybe if I looked more closely, which I won't, because my hand hurts terribly today). A comment would be appropriate indeed.

Reply:
a41f168de7063739d547aaa3e81c7b351056344b